lib/compiler.js

```diff
@@ -440,7 +442,11 @@ function resolveMiddlewarePath(rootDir, middleware) {
   }
 
   if (!fragment) {
-    resolved.sourceFile = require.resolve(pathName);
+    resolvedPath = resolveAppPath(rootDir, middlewarePath);
+    if (resolvedPath === undefined) {
+      throw new Error(pathName);
```

Comment:
Actually a better naming convention is `tryResolveAppPath` for the version that returns `undefined` on when not found, and then `resolveAppPath` that throws an error when not found.